Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableBaseProtocolsFile.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import SwiftSyntax
+import SwiftSyntaxBuilder
+
+let buildableBaseProtocolsFile = SourceFile {
+  ImportDecl(
+    leadingTrivia: .docLineComment(copyrightHeader),
+    path: "SwiftSyntax"
+  )
+
+  // `SyntaxCollectionBuildable` and `ExpressibleAsSyntaxCollectionBuildable` don't exist because of the following reason:
+  // Through `ExpressibleAs*` conformances, a syntax kind might conform to `ExpressibleAsSyntaxCollectionBuildable` via different paths, thus the implementation of `createSyntaxCollectionBuildable` is ambiguous.
+  // We have the same issue for `ExpressibleAsSyntaxBuildable`, but in that case the solution is simple: Create a custom implementaiton of `createSyntaxBuildable` that doesn't perform any of the conversions via the `ExpressibleAs` protocols.
+  // For `SyntaxCollection` we would need to perform at least one conversion to end up with a type that we can put inside a `SyntaxCollection`, so there is no single canonical implementation.
+  // Since the types don't provide any value, we don't generate them for now.
+  for kind in SYNTAX_BASE_KINDS where kind != "SyntaxCollection" {
+    let type = SyntaxBuildableType(syntaxKind: kind)
+    let syntaxType = SyntaxBuildableType(syntaxKind: "Syntax")
+    let isSyntax = type == syntaxType
+    // Types that the `*Buildable` conforms to
+    var buildableConformances: [String] = [type.expressibleAs, type.listBuildable] + (isSyntax ? [] : [syntaxType.listBuildable])
+    var listConformances: [String] = isSyntax ? [] : [syntaxType.buildable]
+
+    ProtocolDecl(
+      modifiers: [TokenSyntax.public],
+      identifier: type.listBuildable,
+      inheritanceClause: listConformances.isEmpty ? nil : TypeInheritanceClause {
+        for conformance in listConformances {
+          InheritedType(typeName: conformance)
+        }
+      }
+    ) {
+      FunctionDecl(
+        leadingTrivia: [
+          "/// Builds list of `\(type.syntax)`s.",
+          "/// - Parameter format: The `Format` to use.",
+          "/// - Parameter leadingTrivia: Replaces the last leading trivia if not nil.",
+        ].map { .docLineComment($0) + .newline }.reduce([], +),
```

Comment:
Generating multi-line doc strings is probably a common pattern that we should move to a utility function. (The reason for not doing a single `.docLineComment` with a multi-line string is that #480 cannot handle indentation correctly in that case IIUC.)